File: DSA/VivaswanSingh_590023022/Day_90/q2.c
#include <stdio.h>

int videoStitching(int clips[][2], int clipsSize, int time)
{
    int currentEnd = 0;
    int farthest = 0;
    int count = 0;

    while (currentEnd < time)
    {
        farthest = currentEnd;

        for (int i = 0; i < clipsSize; i++)
        {
            if (clips[i][0] <= currentEnd && clips[i][1] > farthest)
            {
                farthest = clips[i][1];
            }
        }

        if (farthest == currentEnd)
            return -1;

        currentEnd = farthest;
        count++;
    }

    return count;
}

int main()
{
    int clips[][2] = {
        {0, 2},
        {1, 6},
        {4, 8},
        {7, 10},
        {9, 12}
    };

    int clipsSize = sizeof(clips) / sizeof(clips[0]);
    int time = 10;

    int result = videoStitching(clips, clipsSize, time);

    printf("Minimum clips: %d\n", result);

    return 0;
}

File: DSA/Navneetpant_590011569/Day_90/Q2.java
class Solution {
    public int videoStitching(int[][] clips, int time) {

        Arrays.sort(clips, (a, b) -> {
            if (a[0] == b[0]) {
                return Integer.compare(b[1], a[1]);
            }
            return Integer.compare(a[0], b[0]);
        });

        int count = 0;
        int currentEnd = 0;
        int farthest = 0;
        int i = 0;

        while (currentEnd < time) {

            while (i < clips.length && clips[i][0] <= currentEnd) {
                farthest = Math.max(farthest, clips[i][1]);
                i++;
            }

            if (farthest == currentEnd) {
                return -1;
            }
            count++;
            currentEnd = farthest;
        }

        return count;
    }
}